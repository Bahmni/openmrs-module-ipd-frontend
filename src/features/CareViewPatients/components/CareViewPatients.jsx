import React from "react";
import "../styles/CareViewPatients.scss";
import { Accordion, AccordionItem, Dropdown } from "carbon-components-react";

export const CareViewPatients = () => {
  return (
    <div className="care-view-patients-container">
      <div className="care-view-patients">
        <div className="task-type">
          <Dropdown
            id="default"
            label="Dropdown menu options"
            items={["All tasks", "Pending", "Done"]}
            itemToString={(item) => (item ? item : "")}
            selectedItem={"All tasks"}
          />
        </div>
        <Accordion className="patient-section">
          <AccordionItem title="Patient 1">Patient 1</AccordionItem>
          <AccordionItem title="Patient 2">Patient 2</AccordionItem>
        </Accordion>
      </div>
    </div>
  );
};
