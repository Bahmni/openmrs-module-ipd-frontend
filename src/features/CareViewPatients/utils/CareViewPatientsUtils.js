import axios from "axios";
import {
  BOOKMARK_PATIENT_BASE_URL,
  GET_PATIENT_LIST_URL,
  GET_SEARCH_PATIENT_LIST_URL,
} from "../../../constants";

export const fetchPatientsList = async (wardId, offset, limit = 3) => {
  return await axios.get(GET_PATIENT_LIST_URL.replace("{wardId}", wardId), {
    params: { offset, limit },
  });
};

export const fetchPatientsListBySearch = async (
  wardId,
  searchKeys,
  searchValue,
  offset,
  limit = 3
) => {
  try {
    return await axios.get(
      GET_SEARCH_PATIENT_LIST_URL.replace("{wardId}", wardId),
      {
        params: {
          offset,
          limit,
          searchKeys,
          searchValue,
        },
        paramsSerializer: {
          indexes: null,
        },
      }
    );
  } catch (e) {
    return e;
  }
};

export const bookmarkPatient = async (patientDetails) => {
  return await axios.post(BOOKMARK_PATIENT_BASE_URL, patientDetails);
};
