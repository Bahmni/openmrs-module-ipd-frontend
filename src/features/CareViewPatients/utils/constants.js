export const searchKeys = ["bedNumber", "patientIdentifier", "patientName"];
export const items = [
    { id: "allTasks", text: "All Tasks" },
    { id: "medicationTasks", text: "Medication Tasks" },
    { id: "nonMedicationTasks", text: "Non-Medication Tasks" },
  ];
  