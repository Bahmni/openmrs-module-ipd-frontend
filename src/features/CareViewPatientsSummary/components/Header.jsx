import {
  epochTo12HourTimeFormat,
  epochTo24HourTimeFormat,
} from "../../../utils/DateTimeUtils";
import React, { useContext } from "react";
import propTypes from "prop-types";
import { CareViewContext } from "../../../context/CareViewContext";

export const Header = ({ timeframeLimitInHours, navHourEpoch }) => {
  const { ipdConfig } = useContext(CareViewContext);
  const { enable24HourTime = {} } = ipdConfig;
  return (
    <tr className="patient-row-container">
      <td
        className="patient-details-container"
        key="patient-details-header"
        data-testid="patient-details-header"
      ></td>
      {Array.from({ length: timeframeLimitInHours }, (_, i) => {
        const startTime = navHourEpoch.startHourEpoch + i * 3600;
        if (startTime < navHourEpoch.endHourEpoch) {
          let headerKey = `slot-details-header-${i}`;
          return (
            <td
              className="slot-details-header"
              key={headerKey}
              data-testid={headerKey}
            >
              <div className="time" data-testid={`time-frame-${i}`}>
                {enable24HourTime
                  ? epochTo24HourTimeFormat(startTime, true)
                  : epochTo12HourTimeFormat(startTime, true)}
              </div>
            </td>
          );
        }
      })}
    </tr>
  );
};

Header.propTypes = {
  timeframeLimitInHours: propTypes.number.isRequired,
  navHourEpoch: propTypes.object.isRequired,
};
