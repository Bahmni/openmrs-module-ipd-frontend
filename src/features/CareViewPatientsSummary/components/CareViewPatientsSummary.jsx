import React, { useEffect, useState } from "react";
import propTypes from "prop-types";
import "../styles/CareViewPatientsSummary.scss";
import { HospitalBed16 } from "@carbon/icons-react";
import { Link } from "carbon-components-react";
import { FormattedMessage } from "react-intl";
import { getSlotsForPatientsAndTime } from "../../CareViewSummary/utils/CareViewSummary";

export const CareViewPatientsSummary = (props) => {
  const [setSlotDetails] = useState([]);
  const { patientsSummary } = props;

  const fetchSlots = async (patientUuids, currentTime, endTime) => {
    const response = await getSlotsForPatientsAndTime(
      patientUuids,
      currentTime,
      endTime
    );
    setSlotDetails(response.data);
  };

  useEffect(() => {
    if (patientsSummary.length > 0) {
      const patientUuids = patientsSummary.map(
        (patient) => patient.patientDetails.uuid
      );
      const currentTime = Math.floor(Date.now() / 1000);

      const getTimeInFuture = (offsetHours) => {
        const futureTime = new Date();
        futureTime.setHours(futureTime.getHours() + offsetHours);
        const futureTimeUnix = Math.floor(futureTime.getTime() / 1000);
        return futureTimeUnix.toString();
      };
      const endTime = getTimeInFuture(2);
      fetchSlots(patientUuids, currentTime, endTime);
    }
  }, [patientsSummary]);

  return (
    <div>
      <table className={"care-view-patient-table"}>
        <tbody>
          {patientsSummary.map((patientSummary, idx) => {
            const { patientDetails, bedDetails } = patientSummary;
            const { person } = patientDetails;
            return (
              <tr key={idx} className={"patient-row-container"}>
                <td className={"patient-details-container"}>
                  <div className={"care-view-patient-details"}>
                    <div>
                      <HospitalBed16 />
                      &nbsp;
                      <span>{bedDetails.bedNumber}</span>&nbsp;
                      <Link href={"#"} inline={true}>
                        <span>{patientDetails.display.split(" ")[0]}</span>
                      </Link>
                    </div>
                    <div>
                      <FormattedMessage
                        id={"PATIENT"}
                        defaultMessage={"Patient"}
                      />
                      : <span>{person.display}</span>&nbsp;(
                      <span>{person.gender}</span>)&nbsp;.
                      <span>{person.age}</span>
                      <FormattedMessage
                        id={"AGE_YEARS_LABEL"}
                        defaultMessage={"yrs"}
                      />
                    </div>
                  </div>
                </td>
                <td style={{ minWidth: "650px" }}></td>
                <td style={{ minWidth: "650px" }}></td>
              </tr>
            );
          })}
        </tbody>
      </table>
    </div>
  );
};

CareViewPatientsSummary.propTypes = {
  patientsSummary: propTypes.array,
};
