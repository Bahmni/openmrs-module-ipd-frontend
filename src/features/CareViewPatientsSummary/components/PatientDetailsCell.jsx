import React, { useContext, useEffect, useState } from "react";
import { bookmarkPatient } from "../../CareViewPatients/utils/CareViewPatientsUtils";
import {
  BookmarkAdd20,
  BookmarkFilled20,
  HospitalBed16,
} from "@carbon/icons-react";
import { Link } from "carbon-components-react";
import { FormattedMessage } from "react-intl";
import propTypes from "prop-types";
import PropTypes from "prop-types";
import { CareViewContext } from "../../../context/CareViewContext";
import { getCurrentShiftTimes } from "../../../utils/DateTimeUtils";
import WarningIcon from "../../../icons/warning.svg";
import { getIPDPatientDashboardUrl } from "../../../utils/CommonUtils";

export const PatientDetailsCell = ({
  patientDetails,
  bedDetails,
  careTeamDetails,
  navHourEpoch,
  newTreatments,
  visitDetails,
  previousShiftPendingTasks
}) => {
  const { person, uuid } = patientDetails;
  const {
    ipdConfig,
    provider,
    handleRefreshPatientList,
    handleRefreshSummary,
  } = useContext(CareViewContext);
  const [bookmark, setBookmark] = useState({});
  const { shiftDetails: shiftConfig = {} } = ipdConfig;
  const nurse = bookmark?.provider;
  const formattedNurseName = nurse?.display.includes(" - ")
    ? nurse?.display.split(" - ")[1]
    : nurse?.display;
  const nurseName = nurse && formattedNurseName;
  const isBookmarked = Object.keys(bookmark).length !== 0;

  const getBookmarkStatus = (careTeamDetailsData) => {
    const nearestHourEpochInMilliseconds = navHourEpoch.startHourEpoch * 1000;
    if (!careTeamDetailsData || !careTeamDetailsData.participants) return {};

    for (const participant of careTeamDetailsData.participants) {
      if (
        nearestHourEpochInMilliseconds >= participant.startTime &&
        nearestHourEpochInMilliseconds < participant.endTime
      ) {
        return participant;
      }
    }
    return {};
  };
  const handleBookmarkClick = async (uuid) => {
    const { startDateTime, endDateTime } = getCurrentShiftTimes(shiftConfig);
    const bookmarkData = {
      providerUuid: provider.uuid,
      startTime: startDateTime,
      endTime: endDateTime,
    };

    const unBookmarkData = {
      uuid: bookmark.uuid,
      voided: true,
    };

    const patientData = {
      patientUuid: uuid,
      careTeamParticipantsRequest: [bookmarkData],
    };

    try {
      if (isBookmarked) {
        await bookmarkPatient({
          ...patientData,
          careTeamParticipantsRequest: [unBookmarkData],
        });
        setBookmark({});
        handleRefreshPatientList();
      } else {
        const response = await bookmarkPatient(patientData);
        const participant = getBookmarkStatus(response.data);
        setBookmark(participant);
      }
      handleRefreshSummary();
    } catch (e) {
      handleRefreshPatientList();
    }
  };

  useEffect(() => {
    const participant = getBookmarkStatus(careTeamDetails);
    setBookmark(participant);
  }, []);

  return (
    <td className={"patient-details-container"}>
      <div className={"care-view-patient-details"}>
        <div className={"admission-details"}>
          <HospitalBed16 />|<span>{bedDetails.bedNumber}</span>|
          <Link
            href={getIPDPatientDashboardUrl(
              patientDetails.uuid,
              visitDetails.uuid,
              "careViewDashboard"
            )}
            data-testid="identifier-ipd-dashboard"
          >
            <span>{patientDetails.display.split(" ")[0]}</span>
          </Link>
        </div>
        <div>
          <FormattedMessage id={"PATIENT"} defaultMessage={"Patient"} />:{" "}
          <span>{person.display}</span>&nbsp;(
          <span>{person.gender}</span>)<span className={"separator"}>|</span>
          <span>{person.age}</span>
          <FormattedMessage id={"AGE_YEARS_LABEL"} defaultMessage={"yrs"} />
          {(newTreatments > 0 || previousShiftPendingTasks.length > 0) && (
            <>
              <div className="treatments-notification">
                <div className="warning_icon">
                  <WarningIcon /> </div>
                <div className="treatments-notification-span">
                 { newTreatments > 0 && <div>&bull; { newTreatments + " New treatment(s): "}
                  <>
                    <Link
                      href={getIPDPatientDashboardUrl(
                        patientDetails.uuid,
                        visitDetails?.uuid,
                        "careViewDashboard"
                      )}
                      data-testid="treatments-ipd-dashboard"
                    >
                      <FormattedMessage
                        id={"SCHEDULE_TREATMENTS"}
                        defaultMessage={"Schedule Treatments"}
                      />
                    </Link>
                  </>
                  </div>}
                  { previousShiftPendingTasks.length > 0 &&
                  <div> &bull;{" Previous Pending: "} {
                    previousShiftPendingTasks.map((task, index) => (
                      <span key={task.taskId}>
                        {index === previousShiftPendingTasks.length - 1 ? (
                          <span>{task.taskName}</span>
                        ) : (
                          <span>{task.taskName + ", "}</span>
                        )}
                      </span>
                    ))
                  }</div>
                  }
                </div>
              </div>
            </>
          )}
        </div>
        {isBookmarked && (
          <div>
            <FormattedMessage id={"NURSE"} defaultMessage={"Nurse"} />:{" "}
            <span>{nurseName}</span>
          </div>
        )}
      </div>
      {isBookmarked ? (
        <div
          className={provider.uuid !== nurse?.uuid && "bookmark-disabled"}
          onClick={() => handleBookmarkClick(uuid)}
          data-testid="bookmark-filled-icon"
        >
          <BookmarkFilled20 />
        </div>
      ) : (
        <div
          data-testid="bookmark-add-icon"
          onClick={() => handleBookmarkClick(uuid)}
        >
          <BookmarkAdd20 />
        </div>
      )}
    </td>
  );
};

PatientDetailsCell.propTypes = {
  patientDetails: {
    person: PropTypes.object.isRequired,
    uuid: PropTypes.string.isRequired,
  },
  bedDetails: propTypes.object.isRequired,
  careTeamDetails: propTypes.object.isRequired,
  navHourEpoch: propTypes.object.isRequired,
  visitDetails: propTypes.object.isRequired,
  newTreatments: propTypes.number.isRequired,
};
