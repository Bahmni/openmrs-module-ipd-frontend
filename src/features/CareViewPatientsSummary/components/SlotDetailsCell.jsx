import {
  getColumnData,
  getTaskColumnData,
} from "../../CareViewSummary/utils/CareViewSummary";
import React, { useContext } from "react";
import Clock from "../../../icons/clock.svg";
import { epochTo24HourTimeFormat } from "../../../utils/DateTimeUtils";
import { getAdministrationStatus } from "../../../utils/CommonUtils";
import SVGIcon from "../../SVGIcon/SVGIcon";
import { CareViewContext } from "../../../context/CareViewContext";
import propTypes from "prop-types";

export const SlotDetailsCell = ({
  uuid,
  slotDetails,
  timeframeLimitInHours,
  navHourEpoch,
  nonMedicationDetails,
  filterValue,
}) => {
  const columns = [];
  const { ipdConfig } = useContext(CareViewContext);

  const patientSlotDetail = slotDetails?.find(
    (slotDetail) => slotDetail.patientUuid === uuid
  );

  for (let i = 0; i < timeframeLimitInHours; i++) {
    const startTime = navHourEpoch.startHourEpoch + i * 3600;
    if (startTime < navHourEpoch.endHourEpoch) {
      const endTime = startTime + 3600;
      let columnData, slotColumnData, taskColumnData;
      slotColumnData =
        patientSlotDetail &&
        (filterValue.id === "allTasks" || filterValue.id === "medicationTasks")
          ? patientSlotDetail?.prescribedOrderSlots.flatMap((slot) =>
              getColumnData(slot, startTime, endTime)
            )
          : [];
      taskColumnData =
        nonMedicationDetails &&
        nonMedicationDetails[uuid] &&
        (filterValue.id === "allTasks" ||
          filterValue.id === "nonMedicationTasks")
          ? getTaskColumnData(nonMedicationDetails[uuid], startTime, endTime)
          : [];
      columnData = slotColumnData.concat(...taskColumnData);
      if (columnData) {
        columns.push({ startTime, endTime, columnData });
      } else {
        columns.push({ startTime, endTime, columnData: [] });
      }
    }
  }

  const renderStatusIcon = (slot) => {
    const { drugChart = {} } = ipdConfig;
    const { startTime, status, medicationAdministration } = slot;
    let administrationStatus = getAdministrationStatus(
      medicationAdministration,
      status,
      startTime,
      drugChart,
      slot
    );
    return (
      <div className="status-icon">
        <SVGIcon iconType={administrationStatus} />
      </div>
    );
  };

  const renderColumnData = (columnData) => {
    columnData.sort((a, b) => a.startTime - b.startTime);

    return columnData.map((slotItem) => {
      if (slotItem.isNonMedication) {
        return (
          <div className="slot-details" key={`${slotItem.uuid}`}>
            <div className="logo">
              <div className="status-icon">
                <SVGIcon iconType={"Pending"} />
              </div>
              <Clock className="clock-icon" />
            </div>
            <span>
              {epochTo24HourTimeFormat(slotItem.requestedStartTime / 1000)}
            </span>
            <div className="drug-details-wrapper">
              <span>{slotItem.name}</span>
              <div className="drug-details" data-testid="drug-details">
                {slotItem.creator && (
                  <span className="drug-detail">
                    {slotItem.creator.display}
                  </span>
                )}
              </div>
            </div>
          </div>
        );
      } else {
        const { dose, doseUnits, route } = slotItem.order;
        return (
          <div className="slot-details" key={`${slotItem.uuid}`}>
            <div className="logo">
              {renderStatusIcon(slotItem)}
              <Clock className="clock-icon" />
            </div>
            <span>{epochTo24HourTimeFormat(slotItem.startTime)}</span>
            <div className="drug-details-wrapper">
              <span>{slotItem.order.drug.display}</span>
              <div className="drug-details" data-testid="drug-details">
                {dose && <span className="drug-detail">{dose}</span>}
                {doseUnits && (
                  <span className="drug-detail">{doseUnits?.display}</span>
                )}
                {route && (
                  <span className="drug-detail">
                    {route?.display && `  |  ${route.display}`}
                  </span>
                )}
              </div>
            </div>
          </div>
        );
      }
    });
  };

  return columns.map(({ columnData }, index) => (
    <td className="slot-details-container" key={index}>
      {renderColumnData(columnData)}
    </td>
  ));
};

SlotDetailsCell.propTypes = {
  uuid: propTypes.string.isRequired,
  slotDetails: propTypes.object.isRequired,
  timeframeLimitInHours: propTypes.number.isRequired,
  navHourEpoch: propTypes.object.isRequired,
};
