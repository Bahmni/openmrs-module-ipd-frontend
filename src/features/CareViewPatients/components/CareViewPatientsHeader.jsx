import React from "react";
import PropTypes from "prop-types";
import "../styles/CareViewPatientsHeader.scss";
import { IPD_WARD_SEARCH_PLACEHOLDER_TEXT } from "../../../constants";
import { Dropdown, Search } from "carbon-components-react";

export const CareViewPatientsHeader = (props) => {
  const { handleKeyPress, handleClear, searchValue, updateSearchValue } = props;

  const handleSearchOnChange = (e) => {
    updateSearchValue(e.target.value);
  };

  return (
    <div className="task-type">
      <Search
        placeholder={IPD_WARD_SEARCH_PLACEHOLDER_TEXT}
        size="lg"
        id="ipd-ward-search"
        onChange={handleSearchOnChange}
        value={searchValue}
        onKeyDown={handleKeyPress}
        onClear={handleClear}
        className="ward-patient-search"
        labelText="Patient Search"
        light={true}
      />
      <Dropdown
        id="default"
        label="Dropdown menu options"
        items={["All tasks", "Pending", "Done"]}
        itemToString={(item) => (item ? item : "")}
        selectedItem={"All tasks"}
        light={true}
      />
    </div>
  );
};

CareViewPatientsHeader.propTypes = {
  handleKeyPress: PropTypes.func,
  handleClear: PropTypes.func,
  searchValue: PropTypes.string,
  updateSearchValue: PropTypes.func,
};
