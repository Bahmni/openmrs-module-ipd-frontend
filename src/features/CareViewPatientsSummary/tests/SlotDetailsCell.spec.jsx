import { SlotDetailsCell } from "../components/SlotDetailsCell";
import { mockSlotsData } from "../../CareViewSummary/tests/CareViewSummaryMock";
import React from "react";
import { render, waitFor } from "@testing-library/react";
import { CareViewContext } from "../../../context/CareViewContext";
import { mockConfig } from "../../../utils/CommonUtils";
import "@testing-library/jest-dom/extend-expect";

const mockContext = {
  careViewConfig: { timeframeLimitInHours: 2 },
  ipdConfig: mockConfig,
};

describe("SlotDetailsCell", () => {
  it("should render slotDetailsCell", () => {
    const { container } = render(
      <CareViewContext.Provider value={mockContext}>
        <SlotDetailsCell
          uuid={"3ab51ec0-4650-4400-9aaa-75f30ece0208"}
          slotDetails={mockSlotsData}
          timeframeLimitInHours={2}
          navHourEpoch={{
            startHourEpoch: 1672575400,
            endHourEpoch: 1710511200,
          }}
        />
      </CareViewContext.Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it("renders slot details correctly", async () => {
    const { queryAllByText, queryAllByTestId } = render(
      <CareViewContext.Provider value={mockContext}>
        <SlotDetailsCell
          uuid={"3ab51ec0-4650-4400-9aaa-75f30ece0208"}
          slotDetails={mockSlotsData}
          timeframeLimitInHours={2}
          navHourEpoch={{
            startHourEpoch: 1672575400,
            endHourEpoch: 1710511200,
          }}
        />
      </CareViewContext.Provider>
    );

    await waitFor(() => {
      expect(
        queryAllByText("Amoxicillin/Clavulanic Acid 1000 mg Tablet")
      ).toBeTruthy();
      expect(queryAllByTestId("drug-details")[0]).toHaveTextContent(
        "2Tablet(s) | Oral"
      );
    });
  });

  it("should not render slotDetails data if provided uuid is not present in slotDetails Array", async () => {
    const { queryAllByText, queryAllByTestId } = render(
      <CareViewContext.Provider value={mockContext}>
        <SlotDetailsCell
          uuid={"98a4a9c5-c26e-46a1-8efc-ae6df95842bf"}
          slotDetails={mockSlotsData}
          timeframeLimitInHours={2}
          navHourEpoch={{
            startHourEpoch: 1672575400,
            endHourEpoch: 1710511200,
          }}
        />
      </CareViewContext.Provider>
    );

    await waitFor(() => {
      expect(
        queryAllByText("Amoxicillin/Clavulanic Acid 1000 mg Tablet")
      ).toEqual([]);
      expect(queryAllByTestId("drug-details")).toEqual([]);
    });
  });
});
