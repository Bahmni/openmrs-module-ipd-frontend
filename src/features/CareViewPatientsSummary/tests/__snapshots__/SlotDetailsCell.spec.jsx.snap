// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlotDetailsCell should render slotDetailsCell 1`] = `
<div>
  <td
    class="slot-details-container"
  >
    <div
      class="slot-details"
    >
      <div
        class="logo"
      >
        <div
          class="status-icon"
        >
          <div>
            <test-file-stub />
          </div>
        </div>
        <test-file-stub
          classname="clock-icon"
        />
      </div>
      <span>
        12:16
      </span>
      <div
        class="drug-details-wrapper"
      >
        <span>
          Amoxicillin/Clavulanic Acid 1000 mg Tablet
        </span>
        <div
          class="drug-details"
          data-testid="drug-details"
        >
          <span
            class="drug-detail"
          >
            2
          </span>
          <span
            class="drug-detail"
          >
            Tablet(s)
          </span>
          <span
            class="drug-detail"
          >
              |  Oral
          </span>
        </div>
      </div>
    </div>
  </td>
  <td
    class="slot-details-container"
  />
</div>
`;
